src/table/thead.tsx

```diff
@@ -89,12 +92,30 @@ const Thead = React.forwardRef(
 
     const { columnWidths, totalWidth, updateColumn } = useColumnWidths();
 
+    const stickyStyles = useStickyCellStyles({
```

Comment:
side note/thought: It's unfortunate that we have to use this hook for every Thead, even though it's just used for the select cell. I feel like the select cell should be moved into its own component... 🤔

Reply:
We need it for every sticky thead. Can be optimised as:

```
function Thead(props) {
  if (props.isSticky) {
    return <StickyThead {...props} />
  }
  return <NormalThead {...props} />
}
```

But is it worth it? The hook does not create a subscription for non-sticky cells. Only the refs are assigned.

Reply:
No, look at the whole file: https://github.com/cloudscape-design/components/blob/3de4577329a8eefca3693c2474287eb203d49edf/src/table/thead.tsx.

The `stickyStyles` value is only used for the selection header cell. All other Theads get their sticky state from the `stickyState` prop. 

So maybe we can move this `useStickyCellStyles` call up and pass down the `stickyState` for all Thead cells?

Reply:
Ah, I see. We can't move the useStickyCellStyles up because it produces props for only once cell. Instead, we can create a separate element to the selection cell same as it was done for body cells. However, that's produce quite some diff.